app/src/main/java/com/gdfp/android_evaluationv2/data/TaskAdapter.java

```diff
@@ -92,96 +97,103 @@ public void onBindViewHolder(TaskHolder holder, int position) {
-
+        int indexPriority = mCursor.getColumnIndex(DatabaseContract.TaskColumns.IS_PRIORITY);
 
         //TODO Get the column index of the dueDate
-
+        int indexDueDate = mCursor.getColumnIndex(DatabaseContract.TaskColumns.DUE_DATE);
 
         //TODO Get the column index of the taskDescription
-
+        int indexDescription = mCursor.getColumnIndex(DatabaseContract.TaskColumns.DESCRIPTION);
 
         //TODO Move cursor to position
-
+        mCursor.moveToPosition(position);
 
         //TODO Get the value of id from the cursor
-
+        long id = mCursor.getLong(indexId);
 
         //TODO Get the value of isComplete from the cursor
-
+        int isComplete = mCursor.getInt(indexCheckbox);
 
         //TODO Get the value of priority from the cursor
-
+        int isPriority = mCursor.getInt(indexPriority);
 
         //TODO Get the value of dueDate from the cursor
-
+        long dueDateMillis = mCursor.getLong(indexDueDate);
 
         //TODO Get the value of taskDescription from the cursor
-
+        String description = mCursor.getString(indexDescription);
 
         //TODO Set a tag on the itemView with the id of the Task
-
+        holder.itemView.setTag(id);
 
         //TODO If the Task is done...
-        if (isComplete == 1) {
-
+        if (isComplete== 1) {
             //TODO Check the checkbox
+            holder.checkBox.setChecked(true);
 
             //TODO Set the state of the nameView to "DONE"
-
+            holder.nameView.setState(TaskTitleView.DONE);
 
             //TODO Strike through the text of th nameView
+            holder.nameView.setPaintFlags(Paint.STRIKE_THRU_TEXT_FLAG);
 
             //TODO If the Task is overdue...
-        } else if (dueDate < System.currentTimeMillis()) {
+        } else if (dueDateMillis < System.currentTimeMillis()) {
 
             //TODO Set the state of the nameView to "OVERDUE"
-
+            holder.nameView.setState(TaskTitleView.OVERDUE);
             //TODO un-Check the checkbox
+            holder.checkBox.setChecked(false);
 
 
         } else { //TODO If the Task is normal...
 
             //TODO Set the state of the nameView to "NORMAL"
-
+            holder.nameView.setState(TaskTitleView.NORMAL);
 
             //TODO un-Check the checkbox
-;
+            holder.checkBox.setChecked(false);
+            ;
         }
 
         //TODO If the Task is a priority...
-        if (priority == 1) {
-
+        if (isPriority== 1) {
             //TODO Set the priorityView ImageView with the priority icon
-
+            holder.priorityView.setImageResource(R.drawable.ic_priority);
 
         } else { //TODO If the Task is not a priority...
 
             //TODO Set the priorityView ImageView with the non-priority icon
+            holder.priorityView.setImageResource(R.drawable.ic_not_priority);
 
         }
 
         //TODO If there is no due date...
-        if (dueDate == Long.MAX_VALUE) {
+        if (dueDateMillis == Long.MAX_VALUE) {
```

Comment:
same than above dueDate would be okay, if you have any other scenarios please explain it